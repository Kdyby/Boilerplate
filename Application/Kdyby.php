<?php

/**
 * This file is part of the Framework - Content Managing System (F-CMS) Kdyby.
 *
 * Copyright (c) 2008, 2010 Filip Procházka (http://hosiplan.kdyby.org)
 *
 * For more information please see http://www.kdyby.org
 *
 * @package F-CMS Kdyby-Common
 */



namespace Kdyby\Application;

use Nette;
use Nette\Environment;



final class Kdyby extends Nette\Application\Application
{
	/** @var string */
	public $errorPresenter = 'Error';


}