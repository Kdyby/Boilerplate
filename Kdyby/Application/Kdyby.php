<?php

/**
 * This file is part of the Framework - Content Managing System (CMF) Kdyby.
 *
 * Copyright (c) 2008, 2010 Filip ProchÃ¡zka (http://hosiplan.kdyby.org)
 *
 * For more information please see http://www.kdyby.org
 *
 * @package CMF Kdyby-Common
 */



namespace Kdyby\Application;

use Nette;
use Nette\Environment;



final class Kdyby extends Nette\Application\Application
{
	/** @var string */
	public $errorPresenter = 'Error';



	/**
	 * Dispatch a HTTP request to a front controller.
	 * @return void
	 */
	public function run()
	{
		$this->getContext()->freeze();

		parent::run();
	}



	/**
	 * @return Kdyby\Application\Kdyby
	 */
	public function registerPanels()
	{
		//NetteTranslator\Panel::register(Environment::getService('Nette\ITranslator'), \NetteTranslator\Panel::LAYOUT_VERTICAL);

		//Panel\UserPanel::register()
		//	->addCredentials('HosipLan', 'reddwarf')
		//	->setNameColumn('name');


		// develop environment!
		//\Kdyby\Debug\DoctrinePanel::register();

		return $this;
	}

}