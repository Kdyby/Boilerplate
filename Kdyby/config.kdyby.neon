common:
	php:
		date.timezone: Europe/Prague

	variable:
		varDir: %wwwDir%/../var
		tempDir: %varDir%
		proxiesDir: %varDir%/proxies
		logDir: %varDir%/log
		sessionsDir: %varDir%/sessions
		backupDir: %varDir%/backup
		langDir: %varDir%/lang
		lang: cs_CZ

	Kdyby:
		Core:
			Nette\Security\IIdentity: Kdyby\User\Identity

	service:
		Nette\Application\Application:
			factory: Kdyby\Environment\Configurator::createApplication

		Nette\Application\IPresenterFactory:
			factory: Kdyby\Environment\Configurator::createPresenterFactory
			option:
				context: Â§Nette\IContext

		Nette\Loaders\RobotLoader:
			factory: Kdyby\Environment\Configurator::createRobotLoader
			option:
				directory: [%appDir%, %libsDir%, %proxiesDir%]
			run: true

		Nette\Security\IAuthenticator: Kdyby\Security\Authenticator

		Nette\Web\IUser:
			factory: Kdyby\Environment\Configurator::createIUser

		Nette\Caching\ICacheStorage:
			factory: Kdyby\Environment\Configurator::createCacheStorage
			run: true

		Nette\Caching\IMemcacheStorage:
			factory: Kdyby\Environment\Configurator::createMemcacheStorage
			prefix: Unired.

		Nette\Caching\IMemcacheJournal:
			factory: Kdyby\Environment\Configurator::createMemcacheJournal

		Kdyby\PresenterTree:
			factory: Kdyby\Tools\PresenterTree\PresenterTree::createPresenterTree

		Doctrine\ORM\EntityManager:
			factory: Kdyby\Doctrine\Factory::createEntityManager
			option:
				entityDir: [%appDir%/models, %kdybyDir%/Data, %kdybyDir%/Security]
				proxyDir: %proxiesDir%

		Kdyby\Application\DatabaseManager:
			factory: Kdyby\Environment\Configurator::createDatabaseManager

#		Nette\ITranslator:
#			factory: NetteTranslator\Gettext::getTranslator
#			run: true

		Kdyby\ConfigHooks:
			factory: Kdyby\Environment\Configurator::createConfigHooks


production < common:

development < production:

console < development:
