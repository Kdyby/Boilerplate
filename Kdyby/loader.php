<?php

/**
 * This file is part of the Framework - Content Managing System (CMF) Kdyby.
 *
 * Copyright (c) 2008, 2010 Filip ProchÃ¡zka (http://hosiplan.kdyby.org)
 *
 * This source file is subject to the "Kdyby license", and/or
 * GPL license. For more information please see http://www.kdyby.org
 *
 * @package CMF Kdyby-Common
 */

@header('X-Generated-By: Kdyby CMF ;url=www.kdyby.org'); // @ - headers may be sent

define('KDYBY', TRUE);
define('KDYBY_DIR', __DIR__);

if (!defined('NETTE')) {
	if (!defined('LIBS_DIR')) {
		throw new RuntimeException("Nette Framework cannot be loaded! Missing constant LIBS_DIR");
	}

	// Load Nette Framework
	require_once LIBS_DIR . '/Nette/loader.php';
}

// helper
function cd() {
	foreach (func_get_args() as $arg) {
		\Nette\Debug::barDump($arg);
	}
}

// Configure environment
// enable Nette\Debug for better exception and error visualisation
//Nette\Debug::enable(Nette\Debug::DEVELOPMENT, TEMP_DIR . '/log');
Nette\Debug::enable();
Nette\Debug::$strictMode = TRUE;
//Debug::$maxDepth = 10;
//Debug::$maxLen = 2024;


// register kdyby loader
require_once KDYBY_DIR . '/Loaders/KdybyLoader.php';
Kdyby\Loaders\KdybyLoader::getInstance()->register();


// configure environment
Nette\Environment::setServiceAlias('Doctrine\\ORM\\EntityManager', 'EntityManager');
