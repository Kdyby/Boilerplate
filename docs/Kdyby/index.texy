Kdyby Framework is set of tools providing mainly bridge between an awesome framework "Nette Framework":https://github.com/nette/nette and great persistence layer "Doctrine ORM":https://github.com/doctrine.

Additionaly to this, Kdyby Framework integrates:

- "Assetic":https://github.com/kriswallsmith/assetic for managing css and javascript files and applying filters, allowing you to painlessly use Less or CoffeeScript.
- "Symfony Console":https://github.com/symfony/Console


Chapters:
--------

1. Differences from Nette Framework
   - Form controls
   - Latte Macros
   - Security
2. Packages
3. Persistence & Doctrine
    - Entity Form
5. Migrations
6. Media
7. Curl &Â Browser
8. ACL - RBAC
9. Tools
    - Arrays
    - Objects
    - Strings
    - Filesystem
    - MimeTypeDetector
