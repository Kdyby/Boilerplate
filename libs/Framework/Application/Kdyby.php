<?php

/**
 * This file is part of the Framework - Content Managing System (CMF) Kdyby.
 *
 * Copyright (c) 2008, 2010 Filip ProchÃ¡zka (http://hosiplan.kdyby.org)
 *
 * For more information please see http://www.kdyby.org
 *
 * @package CMF Kdyby-Common
 */



namespace Kdyby\Application;

use Nette;
use Nette\Environment;



final class Kdyby extends Nette\Application\Application
{
	/** @var string */
	public $errorPresenter = 'Error';

	/** @var Kdyby\Injection\IServiceContainer */
	private $container;



	/**
	 * Dispatch a HTTP request to a front controller.
	 * @return void
	 */
	public function run()
	{
		$this->getContainer()->freeze();

		parent::run();
	}



	/**
	 * @return Kdyby\Injection\IServiceContainer
	 */
	public function getContainer()
	{
		if ($this->container === NULL) {
			$this->setContainer(Nette\Environment::getContext());
		}

		return $this->container;
	}



	/**
	 * @return Kdyby\Application\Kdyby
	 */
	public function registerPanels()
	{
		//NetteTranslator\Panel::register(Environment::getService('Nette\ITranslator'), \NetteTranslator\Panel::LAYOUT_VERTICAL);

		//Panel\UserPanel::register()
		//	->addCredentials('HosipLan', 'reddwarf')
		//	->setNameColumn('name');


		// develop environment!
		//\Kdyby\Debug\DoctrinePanel::register();

		return $this;
	}

}