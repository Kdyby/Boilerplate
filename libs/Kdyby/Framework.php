<?php

/**
 * This file is part of the Kdyby (http://www.kdyby.org)
 *
 * Copyright (c) 2008, 2012 Filip Procházka (filip@prochazka.su)
 *
 * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
 */

namespace Kdyby;



/**
 * The Kdyby Framework (http://kdyby.org)
 *
 * @author Filip Procházka <filip@prochazka.su>
 */
final class Framework
{

	const NAME = 'Kdyby Framework';
	const VERSION = '2.0a';
	const REVISION = '$WCREV$ released on $WCDATE$';



	/**
	 * @throws \Kdyby\StaticClassException
	 */
	final public function __construct()
	{
		throw new StaticClassException;
	}



	/**
	 * @return array
	 */
	public static function getDefaultPackages()
	{
		return array(
			'Kdyby\Package\FrameworkPackage\FrameworkPackage',
			'Kdyby\Package\DoctrinePackage\DoctrinePackage',
		);
	}



	/**
	 * @return \Kdyby\Packages\PackagesList
	 */
	public static function createPackagesList()
	{
		return new Packages\PackagesList(static::getDefaultPackages());
	}



	/**
	 * @return array
	 */
	public static function findExceptionClasses()
	{
		return iterator_to_array(\Nette\Utils\Finder::findFiles('exceptions.php')->in(__DIR__));
	}

}
