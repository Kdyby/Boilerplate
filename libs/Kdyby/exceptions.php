<?php

/**
 * This file is part of the Kdyby (http://www.kdyby.org)
 *
 * Copyright (c) 2008, 2012 Filip ProchÃ¡zka (filip@prochazka.su)
 *
 * For the full copyright and license information, please view the file license.md that was distributed with this source code.
 */

namespace Kdyby;



/**
 * The exception that is thrown when the value of an argument is
 * outside the allowable range of values as defined by the invoked method.
 */
class ArgumentOutOfRangeException extends \InvalidArgumentException
{

}



/**
 * The exception that is thrown when a method call is invalid for the object's
 * current state, method has been invoked at an illegal or inappropriate time.
 */
class InvalidStateException extends \RuntimeException
{

}



/**
 * The exception that is thrown when a requested method or operation is not implemented.
 */
class NotImplementedException extends \LogicException
{

}



/**
 * The exception that is thrown when an invoked method is not supported. For scenarios where
 * it is sometimes possible to perform the requested operation, see InvalidStateException.
 */
class NotSupportedException extends \LogicException
{

}



/**
 * The exception that is thrown when a requested method or operation is deprecated.
 */
class DeprecatedException extends NotSupportedException
{

}



/**
 * The exception that is thrown when accessing a class member (property or method) fails.
 */
class MemberAccessException extends \LogicException
{

	/**
	 * @param string $type
	 * @param string|object $class
	 * @param string $property
	 *
	 * @return \Kdyby\MemberAccessException
	 */
	public static function propertyNotWritable($type, $class, $property)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Cannot write to $type property $class::\$$property.");
	}



	/**
	 * @param string|object $class
	 *
	 * @return \Kdyby\MemberAccessException
	 */
	public static function propertyWriteWithoutName($class)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Cannot write to a class '$class' property without name.");
	}



	/**
	 * @param string $type
	 * @param string|object $class
	 * @param string $property
	 *
	 * @return \Kdyby\MemberAccessException
	 */
	public static function propertyNotReadable($type, $class, $property)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Cannot read $type property $class::\$$property.");
	}



	/**
	 * @param string|object $class
	 *
	 * @return \Kdyby\MemberAccessException
	 */
	public static function propertyReadWithoutName($class)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Cannot read a class '$class' property without name.");
	}



	/**
	 * @param string|object $class
	 *
	 * @return MemberAccessException
	 */
	public static function callWithoutName($class)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Call to class '$class' method without name.");
	}



	/**
	 * @param object|string $class
	 * @param string $method
	 *
	 * @return \Kdyby\MemberAccessException
	 */
	public static function undefinedMethodCall($class, $method)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Call to undefined method $class::$method().");
	}

}



/**
 * The exception that is thrown when an I/O error occurs.
 */
class IOException extends \RuntimeException
{

}



/**
 * The exception that is thrown when accessing a file that does not exist on disk.
 */
class FileNotFoundException extends IOException
{

	/**
	 * @param string $file
	 *
	 * @return \Kdyby\FileNotWritableException
	 */
	public static function fromFile($file)
	{
		return new static("Unable to read file '$file'. Please, make this file readable.");
	}

}



/**
 * The exception that is thrown when writing to a file that is not writable.
 */
class FileNotWritableException extends IOException
{

	/**
	 * @param string $file
	 * @return \Kdyby\FileNotWritableException
	 */
	public static function fromFile($file)
	{
		return new static("Unable to write to file '$file'. Please, make this file writable.");
	}

}



/**
 * The exception that is thrown when part of a file or directory cannot be found.
 */
class DirectoryNotFoundException extends IOException
{

	/**
	 * @param string $directory
	 *
	 * @return \Kdyby\DirectoryNotWritableException
	 */
	public static function fromDir($directory)
	{
		return new static("Unable to read directory '$directory'. Please, make this directory readable.");
	}

}



/**
 * The exception that is thrown when writing to a directory that is not writable.
 */
class DirectoryNotWritableException extends IOException
{

	/**
	 * @param string $directory
	 * @return \Kdyby\DirectoryNotWritableException
	 */
	public static function fromDir($directory)
	{
		return new static("Unable to write to directory '$directory'. Please, make this directory writable.");
	}

}



/**
 * The exception that is thrown when an argument does not match with the expected value.
 */
class InvalidArgumentException extends \InvalidArgumentException
{

}



/**
 * The exception that is thrown when an illegal index was requested.
 */
class OutOfRangeException extends \OutOfRangeException
{

}



/**
 * The exception that is thrown when a value (typically returned by function) does not match with the expected value.
 */
class UnexpectedValueException extends \UnexpectedValueException
{


	/**
	 * @param mixed $list
	 * @param string|object $class
	 * @param string $property
	 *
	 * @return \Kdyby\UnexpectedValueException
	 */
	public static function invalidEventValue($list, $class, $property)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Property $class::$$property must be array or NULL, " . gettype($list) . " given.");
	}



	/**
	 * @param string|object $class
	 * @param string $property
	 *
	 * @return \Kdyby\UnexpectedValueException
	 */
	public static function notACollection($class, $property)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Class property $class::\$$property is not an instance of Doctrine\\Common\\Collections\\Collection.");
	}



	/**
	 * @param string|object $class
	 * @param string $property
	 *
	 * @return \Kdyby\UnexpectedValueException
	 */
	public static function collectionCannotBeReplaced($class, $property)
	{
		$class = is_object($class) ? get_class($class) : $class;
		return new static("Class property $class::\$$property is an instance of Doctrine\\Common\\Collections\\Collection. Use add<property>() and remove<property>() methods to manipulate it or declare your own.");
	}

}



/**
 * The exception that is thrown when static class is instantiated.
 */
class StaticClassException extends \LogicException
{

}
