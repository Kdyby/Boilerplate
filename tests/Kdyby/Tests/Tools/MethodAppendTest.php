<?php

/**
 * This file is part of the Kdyby (http://www.kdyby.org)
 *
 * Copyright (c) 2008, 2012 Filip Procházka (filip@prochazka.su)
 *
 * For the full copyright and license information, please view the file license.md that was distributed with this source code.
 */

namespace Kdyby\Tests\Tools;

use Kdyby;
use Kdyby\Tools\MethodAppend;
use Nette;
use Nette\Reflection\ClassType;
use Nette\PhpGenerator as Code;
use Nette\Utils\Strings;



/**
 * @author Filip Procházka <filip@prochazka.su>
 */
class MethodAppendTest extends Kdyby\Tests\TestCase
{

	/**
	 * @return \Nette\Reflection\ClassType
	 */
	private function prepareClass()
	{
		$tempDir = $this->getContext()->expand('%tempDir%/classes');
		Kdyby\Tools\Filesystem::mkDir($tempDir);

		$class = new Code\ClassType('MyClass_' . Strings::random());
		$foo = $class->addMethod('foo')
			->addBody('$c = $a + $b;');
		$foo->addParameter('a');
		$foo->addParameter('b');

		$class->addMethod('bar')
			->addBody('return $this->foo(1, 2);');

		$file = $tempDir . '/' . $class->name . '.class.php';
		file_put_contents($file, '<?php'. "\n\n" . (string)$class);
		require_once $file;

		return ClassType::from($class->name);
	}



	/**
	 * @param string $name
	 * @param string $className
	 *
	 * @return string
	 */
	private function expected($name, $className)
	{
		$expected = file_get_contents(__DIR__ . "/Fixtures/MethodAppendTest.$name.expected");
		return strtr($expected, array(
			'<generated_class_name>' => $className
		));
	}



	public function testAppend()
	{
		$class = $this->prepareClass();
		$method = new MethodAppend($class->getMethod('foo'));
		$method->append('$myCode = "lipsum";');

		$result = file_get_contents($class->getFileName());
		$this->assertEquals($this->expected('functional1', $class->name), $result);
	}



	public function testAppend_Multiple()
	{
		$class = $this->prepareClass();
		$method = new MethodAppend($class->getMethod('foo'));
		$method->append('$myCode = "lipsum";');
		$method->append('$myCode = "dolor";');
		$method->append('$myCode = "sit amet";');

		$result = file_get_contents($class->getFileName());
		$this->assertEquals($this->expected('functional2', $class->name), $result);
	}

}
