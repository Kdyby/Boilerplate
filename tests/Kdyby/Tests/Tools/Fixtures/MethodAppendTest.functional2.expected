<?php

class <generated_class_name>
{

	public function foo($a, $b)
	{
		$c = $a + $b;

		$myCode = "lipsum";

		$myCode = "dolor";

		$myCode = "sit amet";
	}


	public function bar()
	{
		return $this->foo(1, 2);
	}

}
